win32ss/gdi/ntgdi/text.c

```diff
@@ -13,6 +13,48 @@
 #define NDEBUG
 #include <debug.h>
 
+
+/*
+   This is a hack. See CORE-1091.
+
+   It is needed because ReactOS does not support raster fonts now
+   After Raster Font support is added, then it can be removed
+   Find the current font's logfont for testing its lf.lfFaceName
+
+   The ftGdiGetTextMetricsW function currently in ReactOS will always return a Truetype font
+   because we cannot yet handle raster fonts. So it will return flags
+   TMPF_VECTOR and TMPF_TRUETYPE, which can cause problems in edit boxes
+ */
+
+VOID FASTCALL
+IntTMWFixUp(
+   HDC hDC,
+   TMW_INTERNAL *ptm )
+{
+    LOGFONTW lf;
+    HFONT hCurrentFont;
+
+    hCurrentFont = NtGdiGetDCObject( hDC, GDI_OBJECT_TYPE_FONT );
+    GreGetObject(hCurrentFont, sizeof(LOGFONTW), &lf );
+
+    /* To compensate for the GetTextMetricsW call changing the PitchAndFamily */
+    /* to a Truetype one when we have a 'helv' font as our input we and (&)   */
+    /* our current PitchAndFamily with 0xF9 to remove the two problem bits    */
+    /* Out list below checks for Raster Font Facenames                        */
```

Comment:
```suggestion
    /* To compensate for the GetTextMetricsW call changing the PitchAndFamily
     * to a TrueType one when we have a 'helv' font as our input we and (&)
     * our current PitchAndFamily with 0xF9 to remove the two problem bits.
     * Out list below checks for Raster Font Facenames. */
```

Reply:
"we and (&) [...] with 0xF9 to remove the two problem bits."
==> "we filter out the problematic TrueType and Vector bits."